bindings/java/src/org/sleuthkit/datamodel/IngestJobInfo.java

```diff
@@ -0,0 +1,204 @@
+		this.ingestJobId = ingestJobId;
+		this.dataSourceId = dataSourceId;
+		this.hostName = hostName;
+		this.startDateTime = startDateTime;
+		this.settingsDir = settingsDir;
+		this.skCase = skCase;
+		this.ingestModuleInfo = ingestModuleInfo;
+		this.status = IngestJobStatusType.STARTED;
+	}
+
+	/**
+	 * Constructs an IngestJobInfo that has already ended
+	 *
+	 * @param ingestJobId      The id of the ingest job
+	 * @param dataSourceId     The data source the job is being run on
+	 * @param hostName         The host on which the job was executed
+	 * @param startDateTime    The date time the job was started
+	 * @param endDateTime      The date time the job was ended (if it ended)
+	 * @param settingsDir      The directory of the job settings
+	 * @param ingestModuleInfo The ingest modules being run for this job
+	 * @param skCase           A reference to sleuthkit case
+	 */
+	IngestJobInfo(int ingestJobId, long dataSourceId, String hostName, Date startDateTime, Date endDateTime, IngestJobStatusType status, String settingsDir, List<IngestModuleInfo> ingestModuleInfo, SleuthkitCase skCase) {
+		this.ingestJobId = ingestJobId;
+		this.dataSourceId = dataSourceId;
+		this.hostName = hostName;
+		this.startDateTime = startDateTime;
+		this.endDateTime = endDateTime;
+		this.settingsDir = settingsDir;
+		this.skCase = skCase;
+		this.ingestModuleInfo = ingestModuleInfo;
+		this.status = status;
+	}
+
+	/**
+	 * @return the end date time of the job (equal to the epoch if it has not
+	 *         been set yet).
+	 */
+	public Date getEndDateTime() {
+		return endDateTime;
+	}
+
+	/**
+	 * Sets the end date for the ingest job info, and updates the database.
+	 *
+	 * @param endDate the endDateTime to set
+	 *
+	 * @throws org.sleuthkit.datamodel.TskCoreException If the update fails.
+	 * @throws org.sleuthkit.datamodel.TskDataException If the job has already
+	 *                                                  been given an end date
+	 *                                                  or is not in the
+	 *                                                  database.
+	 */
+	public void setEndDateTime(Date endDateTime) throws TskCoreException, TskDataException {
+		this.endDateTime = endDateTime;
+		try {
+			skCase.setIngestJobEndDateTime(getIngestJobId(), endDateTime.getTime());
+		} catch (TskCoreException ex) {
+			this.endDateTime = new Date(0);
+			throw ex;
+		} catch (TskDataException ex) {
+			this.endDateTime = new Date(0);
+			throw ex;
+		}
+	}
+
+	/**
+	 * Sets the ingest status for the ingest job info, and updates the database.
+	 *
+	 * @param status The new status
+	 *
+	 * @throws TskCoreException If the update fails
+	 * @throws TskDataException If the job is not in the db.
+	 */
+	public void setIngestJobStatus(IngestJobStatusType status) throws TskCoreException, TskDataException {
```

Comment:
We really don't need the TskDataException and documentation; the objects are constructed privately by the the SleuthkitCase class and should be in the database. Simplifying the exception specification makes life a little easier for clients.